src/frontends/tensorflow_common/src/op/size.cpp

```diff
@@ -19,12 +21,31 @@ namespace op {
 
 ov::OutputVector translate_size_op(const NodeContext& node) {
     // Size operation computes a number of elements in the input tensor
-    default_op_checks(node, 1, {"Size"});
+    default_op_checks(node, 1, {"Size"}, true);
     auto input = node.get_input(0);
 
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+
     // retrive attribute of the output type
     auto out_type = node.get_attribute<element::Type>("out_type", element::i32);
 
+    if (complex_type_mark) {
+
+        input = complex_type_mark->input_value(0);
+
+        // compute the input tensor size
+        auto shape_of = make_shared<v3::ShapeOf>(input, out_type);
+        auto axis = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto complex_size = make_shared<v1::ReduceProd>(shape_of, axis);
+
+        // Since I'm handling complex numbers, needed to divide the size by 2 because real and imaginary parts are counted separately
+        auto complex_size_divided_by_two = 
+            make_shared<v1::Divide>(complex_size, make_shared<v0::Constant>(element::i32, Shape{}, 2));
+
+        set_node_name(node.get_name(), complex_size_divided_by_two);
+        return {complex_size_divided_by_two->output(0)};
```

Comment:
```suggestion
        input = complex_type_mark->input_value(0);

        // compute the input tensor size
        auto shape_of = make_shared<v3::ShapeOf>(input, out_type);
        auto axis = make_shared<v0::Constant>(element::i32, Shape{}, 0);
        auto complex_size = make_shared<v1::ReduceProd>(shape_of, axis);

        // need to divide the size by 2 because real and imaginary parts are counted separately
        auto complex_size_divided_by_two = 
            make_shared<v1::Divide>(complex_size, make_shared<v0::Constant>(element::i32, Shape{}, 2));

        set_node_name(node.get_name(), complex_size_divided_by_two);
        return {complex_size_divided_by_two->output(0)};
```